common/browser_process_test.go

```diff
@@ -0,0 +1,151 @@
+package common
+
+import (
+	"context"
+	"io"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type mockReader struct {
+	lines []string
+	err   error
+}
+
+func (r mockReader) Read(p []byte) (n int, err error) {
+	for _, l := range r.lines {
+		n += copy(p, []byte(l+"\n"))
+	}
+	return n, r.err
+}
```

Comment:
Thanks, you're right :+1: I didn't check what `Scanner.Text()` returned, and just assumed that `Scan()` would tokenize whatever `Read()` returned in a single call. :facepalm:

I pushed the change in 024381d, with some minor tweaks. Mainly, returning `io.EOF` to avoid the `multiple Read calls return no data or error` error.

But I'm not sure what you're suggesting with that previous code. Let's discuss the refactor in the other thread.

Reply:
> I'm not sure what you're suggesting with that previous code

The previous code I shared is the one on the current `main`. I just added `<-cmd.done` to it to make it work with the test in this PR. It wasn't about refactoring.